lib/operations/add_user.js

```diff
@@ -1,184 +1,173 @@
 'use strict';
 
-const OperationBase = require('./operation').OperationBase;
+const Aspect = require('./operation').Aspect;
 const CommandOperation = require('./command');
+const defineAspects = require('./operation').defineAspects;
 const applyWriteConcern = require('../utils').applyWriteConcern;
 const CONSTANTS = require('../constants');
 const count = require('./collection_ops').count;
 const crypto = require('crypto');
 const handleCallback = require('../utils').handleCallback;
 const loadDb = require('../dynamic_loaders').loadDb;
-const MongoError = require('mongodb-core').MongoError;
-const ReadPreference = require('mongodb-core').ReadPreference;
 const toError = require('../utils').toError;
 const updateOne = require('./collection_ops').updateOne;
+const WriteConcern = require('../write_concern');
 
-class AddUserOperation extends OperationBase {
+class AddUserOperation extends CommandOperation {
   constructor(db, username, password, options) {
-    super(options);
+    options = options || {};
 
-    this.db = db;
-    this.username = username;
-    this.password = password;
-  }
-
-  execute(callback) {
-    const db = this.db;
-    const username = this.username;
-    const password = this.password;
-    const options = this.options;
+    // Special case where there is no password ($external users)
+    if (typeof username === 'string' && password != null && typeof password === 'object') {
+      options = password;
+      password = null;
+    }
 
-    let Db = loadDb();
+    // Get additional values
+    //const customData = options.customData != null ? options.customData : {};
+    let roles = Array.isArray(options.roles) ? options.roles : [];
 
-    // Did the user destroy the topology
-    if (db.serverConfig && db.serverConfig.isDestroyed()) {
-      return callback(new MongoError('topology was destroyed'));
+    // If not roles defined print deprecated message
+    if (roles.length === 0) {
+      console.log('Creating a user without roles is deprecated in MongoDB >= 2.6');
```

Comment:
Let's add a TODO to deal with this. I don't like that this is just a normal `console.log`